src/util/git-ref-resolver.js

```diff
@@ -38,39 +39,36 @@ class GitRefResolver {
 
   async resolve(): Promise<DefaultBranch | ResolvedSha | false> {
     const testFunctions = [
```

Comment:
Btw. I'm not very convinced that we need a class for this. Just having these functions in the module may be a more efficient approach. This is up to you though, just sharing my observation.